src/adapter/src/catalog/migrate.rs

```diff
@@ -812,3 +813,40 @@ fn catalog_fix_system_cluster_replica_ids_v_0_95_0(
     }
     Ok(())
 }
+
+/// This migration applies the rename of the built-in `mz_introspection` cluster to
+/// `mz_catalog_server` to the durable catalog state.
+fn catalog_rename_mz_introspection_cluster_v_0_102_0(
+    tx: &mut Transaction,
+    boot_ts: EpochMillis,
+) -> Result<(), anyhow::Error> {
+    use mz_audit_log::{EventDetails, EventType, ObjectType, RenameClusterV1, VersionedEvent};
+
+    let found_cluster_id = tx
+        .get_clusters()
+        .find(|cluster| cluster.id.is_system() && cluster.name == "mz_introspection")
+        .map(|cluster| cluster.id);
+    let Some(cluster_id) = found_cluster_id else {
+        return Ok(());
+    };
+
+    tx.rename_cluster(cluster_id, "mz_introspection", "mz_catalog_server")?;
+
+    // Update audit log.
+    let audit_id = tx.allocate_audit_log_id()?;
+    let rename_event = VersionedEvent::new(
+        audit_id,
+        EventType::Alter,
+        ObjectType::Cluster,
+        EventDetails::RenameClusterV1(RenameClusterV1 {
+            id: cluster_id.to_string(),
+            old_name: "mz_introspection".into(),
+            new_name: "mz_catalog_server".into(),
+        }),
+        None,
+        boot_ts,
+    );
+    tx.insert_audit_log_event(rename_event);
```

Comment:
Do we want to emit an audit log event for this rename? On one hand it is a bit weird, since AFAIK we don't emit `create` events for system clusters. On the other hand, the migration above emits audit events for changing the IDs of system cluster replicas, which is a very similar operation.

Reply:
Although we do not emit the initial create events, I think it's overall good to issue this rename event. I looked through notion and `fn add_new_builtin_clusters_migration` and was unable to find any notes about reasoning to exclude those events in the first place. So let's move ahead with issuing this rename audit event even though it doesn't have a create counterpart.